### PR TITLE
docs: app deployments (persisted documents)

### PR DESCRIPTION
**Pull request by n1ru4l**
### Background

documentation for app deployments

### Checklist

- [ ] Input validation
- [ ] Output encoding
- [ ] Authentication management
- [ ] Session management
- [ ] Access control
- [ ] Cryptographic practices
- [ ] Error handling and logging
- [ ] Data protection
- [ ] Communication security
- [ ] System configuration
- [ ] Database security
- [ ] File management
- [ ] Memory management
- [ ] Testing

### CODE REVIEW

**Review by dotansimha — APPROVED** [review]

File: packages/web/docs/src/pages/product-updates/2024-07-30-persisted-documents-app-deployments-preview.mdx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import NextImage from 'next/image'
+
+**TL;DR** Persisted documents through app deployments are now available in preview on Hive and allow
+you to secure your GraphQL API. To get started, please refer to the
+[app deployments documentation](/docs/features/app-deployments).
+
+## What are Persisted Documents?
+
+Persisted documents (also known as persisted queries or trusted documents) are a way to pre-register
```
Comment (saihaj) [inline packages/web/docs/src/pages/product-updates/2024-07-30-persisted-documents-app-deployments-preview.mdx:18]: should we link to some more docs on these terms?
```suggestion
Persisted documents (also known as [persisted queries](https://relay.dev/docs/guides/persisted-queries/) or [trusted documents](https://benjie.dev/graphql/trusted-documents)) are a way to pre-register
```

File: packages/web/docs/src/pages/product-updates/2024-07-30-persisted-documents-app-deployments-preview.mdx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    B-- "Load persisted
+ documents" --> C
+    C-- "Report persisted
+ document usage" -->B
+    D-- "Send persisted
+ document request" -->C
+```
+
+Furthermore, you get analytics specific to your deployed app version, so you can make better
+decissions on deprecating and removing fields from your GraphQL API schema and avoid breaking
```
Comment (saihaj) [inline packages/web/docs/src/pages/product-updates/2024-07-30-persisted-documents-app-deployments-preview.mdx:87]:
```suggestion
decisions on deprecating and removing fields from your GraphQL API schema and avoid breaking
```

File: packages/web/docs/src/pages/docs/features/app-deployments.mdx [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+| --------------- | ----------------------------------------------------------------------- |
+| 200             | The persisted document was found and is returned in the response body.  |
+| 404             | The persisted document does not exist.                                  |
+| 401             | The request is unauthorized. A valid CDN access token must be provided. |
+
+</Tabs.Tab>
+
+</Tabs>
+
+## Sending Persisted Document Requests from your App
```
Comment (saihaj) [inline packages/web/docs/src/pages/docs/features/app-deployments.mdx:313]: I think we should add docs for GraphQL Client specific integrations. Similar to what is being done for integration on servers we should have for clients

File: packages/web/docs/src/pages/product-updates/2024-07-30-persisted-documents-app-deployments-preview.mdx (hunk 1 above)
Comment (n1ru4l) [reply packages/web/docs/src/pages/product-updates/2024-07-30-persisted-documents-app-deployments-preview.mdx:18]: I would not link to relay docs, they don't provide anything useful information on why

File: packages/web/docs/src/pages/docs/features/app-deployments.mdx (hunk 3 above)
Comment (n1ru4l) [reply packages/web/docs/src/pages/docs/features/app-deployments.mdx:313]: agreed, lets do it in a follow-up PR

**Pull request merged**